waku/v2/node/wakuswitch.nim

```diff
@@ -14,19 +18,21 @@ proc withWsTransport*(b: SwitchBuilder): SwitchBuilder =
   b.withTransport(proc(upgr: Upgrade): Transport = WsTransport.new(upgr))
 
 proc getSecureKey(path : string): TLSPrivateKey
-  {.raises: [Defect,TLSStreamProtocolError, IOError].} =
+  {.raises: [Defect, IOError].} =
```

Comment:
Perhaps out of scope since it was already there, but if `TLSPrivateKey.init(stringkey)` fails, we will be returning an uninitialised `TLSPrivateKey` and since we don't track `isErr()` nor raise an exception, no one will notice, with perhaps unexpected behaviours somewhere else.

Are we tracking this issue somewhere?

Is it only me, or these Nim implicit empty returns can be dangerous?

```nim
proc foo(inArg: string): int =
  if inArg == "hi":
    return 100

echo foo("hola") # outputs:0
```

Reply:
True, with a submodule update I try to only make the absolute necessary changes.

Have started a laundry list of error handling to improve:
https://github.com/status-im/nwaku/issues/1312